net/redisclient.go

```diff
@@ -25,6 +26,16 @@ import (
 type RedisOptions struct {
 	// Addrs are the list of redis shards
 	Addrs []string
+
+	// AddrUpdater is a func that is regularly called to update
+	// redis address list. This func should return a list of redis
+	// shards.
+	AddrUpdater func() []string
+
+	// UpdateInterval is the time.Duration that AddrUpdater is
+	// triggered and SetAddrs be used to update the redis shards
+	UpdateInterval time.Duration
```

Comment:
As an idea: alternatively we can have a channel for updates `chan []string` and update addresses when new values arrive. This way external updater function can either send updates at regular intervals or it may be a k8s watch/informer that would only send updates when endpoints change.